tools/travis/run_tests.sh

```diff
@@ -20,48 +20,67 @@
 
 set -ev
 
-if [ "$#" -ne 1 ] ; then
-    echo "Usage: "$0" <test>"
+argc="$#"
+if [ "$argc" -gt 2 ]; then
+    echo "Usage: "$0" <test> [<use-run>]"
     exit 1
 fi
 
-test="$1"
+test_suite="$1"
+use_run="$2"
 
-pattern="$2"
 
-if [ "$test" = "static" ]; then
+if [ "$TRAVIS_OS_NAME" = "osx" ] && [ -z "$SNAPCRAFT_FROM_BREW" ]; then
+    echo "osx tests require snapcraft from brew to be setup"
+    exit 0
+fi
+
+if [ "$TRAVIS_OS_NAME" = "osx" ]; then
+    brew upgrade python
+    # readlink -f doesn't work on osx.
```

Comment:
I'm guessing this comment is misplaced here  and belongs further down where greadlink is being used?